RPiNWR/messages/SAME.py

```diff
@@ -248,7 +262,7 @@ def _truncate(avgmsg, confidences):
     for i in range(0, len(avgmsg)):
         if frame[i] != '_':
             if avgmsg[i] != frame[i]:
-                avgmsg[i] = frame[i]
+                avgmsg = mutate_string(avgmsg, i, frame[i])
```

Comment:
How does this pattern compare to, before the loop (or upon the first change) creating an array and afterwards creating a string from that array?